test/test-signal.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// Testcase to produce NodeReport on signal
+if (process.argv[2] === 'child') {
+  require('../').setEvents("signal");
+
+  function myLoop() {
+  var list = [];
+    for (var i=0; i<10000000000; i++) {
+      for (var j=0; i<1000; i++) {
+        list.push(new MyRecord());
+      }
+      for (var j=0; i<1000; i++) {
+        list[j].id += 1;
+        list[j].account += 2;
+      }
+      for (var j=0; i<1000; i++) {
+        list.pop();
+      }
+    }
+  }
+
+  function MyRecord() {
+    this.name = 'foo';
+    this.id = 128;
+    this.account = 98454324;
+  }
+
+  myLoop();
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  if (process.platform === 'win32') {
+    tap.plan(0, 'Signals not supported on Windows');
+    process.exit(0);
+  }
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  setTimeout(() => {
+    child.kill('SIGUSR2');
+    setTimeout(() => {
+      child.kill('SIGTERM');
+    }, 1000);
+  }, 1000);
```

Comment:
timeouts are touchy in tests. on arm, it may take more than a second to start the child, and then this will break

better to wait asynchronously for the events you depend on. I would fork the child, not spawn it, or spawn it with an ipc channel and have the child send a msg to the parent when it is ready

then I would have the parent loop, looking at the filesystem, waiting for `common.locate()` to return a file

this might run a lot faster in the normal case, and also pass on really slow systems